packages/language-support/src/syntaxValidation/syntaxValidation.ts

```diff
@@ -312,3 +356,25 @@ function warnOnUndeclaredFunctions(
 
   return warnings;
 }
+
+function errorOnUndeclaredProcedures(
+  parsingResult: ParsedStatement,
+  dbSchema: DbSchema,
+): SyntaxDiagnostic[] {
+  const errors: SyntaxDiagnostic[] = [];
+
+  if (dbSchema.procedures) {
+    const proceduresInQuery = parsingResult.collectedProcedures;
+
+    proceduresInQuery.forEach((parsedProcedure) => {
+      const error = detectNonDeclaredProcedure(
+        parsedProcedure,
+        dbSchema.procedures,
+      );
+
+      if (error) errors.push(error);
+    });
```

Comment:
The extra layer of `detectNonDeclaredProcedure` isn't very useful imo, I think it's easier to understand and better to inline it. That way we can also remove one of the if statements:

```suggestion
    proceduresInQuery.forEach((parsedProcedure) => {
      const procedureExists = Boolean(proceduresSchema[parsedProcedure.name]);
      if (!procedureExists) {
         errors.push(generateProcedureNotFoundError(parsedProcedure))
      }
}
```

Reply:
done